bus-mapping/src/evm/opcodes/sstore.rs

```diff
@@ -0,0 +1,176 @@
+use super::Opcode;
+use crate::circuit_input_builder::{CircuitInputStateRef, ExecStep};
+use crate::operation::{CallContextField, CallContextOp, TxRefundOp};
+use crate::{
+    operation::{StorageOp, TxAccessListAccountStorageOp, RW},
+    Error,
+};
+
+use eth_types::{GethExecStep, ToWord, Word};
+
+/// Placeholder structure used to implement [`Opcode`] trait over it
+/// corresponding to the [`OpcodeId::SSTORE`](crate::evm::OpcodeId::SSTORE)
+/// `OpcodeId`.
+#[derive(Debug, Copy, Clone)]
+pub(crate) struct Sstore;
+
+impl Opcode for Sstore {
+    fn gen_associated_ops(
+        state: &mut CircuitInputStateRef,
+        geth_steps: &[GethExecStep],
+    ) -> Result<Vec<ExecStep>, Error> {
+        let geth_step = &geth_steps[0];
+        let mut exec_step = state.new_step(geth_step)?;
+
+        let _call_id = state.call()?.call_id;
+        let contract_addr = state.call()?.address;
+
+        state.push_op(
+            &mut exec_step,
+            RW::READ,
+            CallContextOp {
+                call_id: state.call()?.call_id,
+                field: CallContextField::TxId,
+                value: Word::from(state.tx_ctx.id()),
+            },
+        );
+        state.push_op(
+            &mut exec_step,
+            RW::READ,
+            CallContextOp {
+                call_id: state.call()?.call_id,
+                field: CallContextField::RwCounterEndOfReversion,
+                value: Word::from(state.call()?.rw_counter_end_of_reversion),
+            },
+        );
+        state.push_op(
+            &mut exec_step,
+            RW::READ,
+            CallContextOp {
+                call_id: state.call()?.call_id,
+                field: CallContextField::IsPersistent,
+                value: Word::from(state.call()?.is_persistent as u8),
+            },
+        );
+        state.push_op(
+            &mut exec_step,
+            RW::READ,
+            CallContextOp {
+                call_id: state.call()?.call_id,
+                field: CallContextField::CalleeAddress,
+                value: state.call()?.address.to_word(),
+            },
+        );
+
+        let key = geth_step.stack.nth_last(0)?;
+        let key_stack_position = geth_step.stack.nth_last_filled(0);
+        let value = geth_step.stack.nth_last(1)?;
+        let value_stack_position = geth_step.stack.nth_last_filled(1);
+
+        state.push_stack_op(&mut exec_step, RW::READ, key_stack_position, key)?;
+        state.push_stack_op(&mut exec_step, RW::READ, value_stack_position, value)?;
+
+        let is_warm = state
+            .sdb
+            .check_account_storage_in_access_list(&(contract_addr, key));
+
+        let (_, value_prev) = state.sdb.get_storage(&contract_addr, &key);
+        let value_prev = *value_prev;
+        let (_, committed_value) = state.sdb.get_committed_storage(&contract_addr, &key);
+        let committed_value = *committed_value;
```

Comment:
```suggestion
        let value_prev = value_prev.clone();
        let (_, committed_value) = state.sdb.get_committed_storage(&contract_addr, &key);
        let committed_value = committed_value.clone();
```

Make the clones explicit please.